src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -210,4 +220,55 @@ private CommandResult executeCommand(String commandText) throws CommandException
             throw e;
         }
     }
+
+    /**
+     * Displays the top three upcoming appointments for all persons in the system
+     *
+     * @see Person
+     * @see Schedule
+     */
+    private void showTopThreeSchedules() {
+        List<Map.Entry<Person, LocalDateTime>> allAppointments = new ArrayList<>();
+
+        for (Person person : personList) {
+            for (Schedule schedule : person.getSchedules()) {
+                if (schedule.getDateTime().isEmpty()) {
+                    continue;
```

Comment:
Is this a bit too nested? Would it be better to abstract the code out to another method?